src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseDeprecatedSecurityProtocols.cs

```diff
@@ -61,47 +61,21 @@ public override void Initialize(AnalysisContext context)
                         return;
                     }
 
-                    bool IsReferencingSecurityProtocolType(
-                        IFieldReferenceOperation fieldReferenceOperation,
-                        out bool isDeprecatedProtocol,
-                        out bool isHardCodedOkayProtocol)
-                    {
-                        if (securityProtocolTypeTypeSymbol.Equals(fieldReferenceOperation.Field.ContainingType))
-                        {
-                            if (HardCodedSafeProtocolMetadataNames.Contains(fieldReferenceOperation.Field.Name))
-                            {
-                                isHardCodedOkayProtocol = true;
-                                isDeprecatedProtocol = false;
-                            }
-                            else if (fieldReferenceOperation.Field.Name == SystemDefaultName)
-                            {
-                                isHardCodedOkayProtocol = false;
-                                isDeprecatedProtocol = false;
-                            }
-                            else
-                            {
-                                isDeprecatedProtocol = true;
-                                isHardCodedOkayProtocol = false;
-                            }
-
-                            return true;
-                        }
-                        else
-                        {
-                            isHardCodedOkayProtocol = false;
-                            isDeprecatedProtocol = false;
-                            return false;
-                        }
-                    }
-
                     compilationStartAnalysisContext.RegisterOperationAction(
                         (OperationAnalysisContext operationAnalysisContext) =>
                         {
                             var fieldReferenceOperation = (IFieldReferenceOperation)operationAnalysisContext.Operation;
+
+                            // Make sure we're not inside an &= assignment like:
```

Comment:
Long term, do we want to consider doing value content dataflow analysis (assuming it handles all the involved operators, otherwise we can add support)?

Reply:
Unless value content DFA could track bits being enabled / disabled (for people using |= assignments), I don't think it'd be worth it for these particular rules.

But now I'm thinking I should make sure I'm excluding &= assignments for ServicePointManager.SecurityProtocol.

Reply:
I am not sure we have tests, but the code seems to be there to handle binary OR and binary AND operations: https://github.com/dotnet/roslyn-analyzers/blob/558ca610c8f6d67ae091ecf3b57951a08b2b2d61/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ValueContentAnalysis/ValueContentAbstractValue.cs#L430-L442

https://github.com/dotnet/roslyn-analyzers/blob/558ca610c8f6d67ae091ecf3b57951a08b2b2d61/src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ValueContentAnalysis/ValueContentAnalysis.ValueContentDataFlowOperationVisitor.cs#L196-L204

Reply:
That only works if DFA knows the initial value, right?  If someone just has a method that only contains `ServicePointManager.SecurityProtocol |= SecurityProtocolType.Ssl3`, we'd still end up with Unknown, wouldn't we?

Reply:
Ah, I get your point...